base/client.jl

```diff
@@ -84,35 +84,65 @@ answer_color() = text_colors[repl_color("JULIA_ANSWER_COLOR", default_color_answ
 stackframe_lineinfo_color() = repl_color("JULIA_STACKFRAME_LINEINFO_COLOR", :bold)
 stackframe_function_color() = repl_color("JULIA_STACKFRAME_FUNCTION_COLOR", :bold)
 
-function repl_cmd(cmd, out)
-    shell = shell_split(get(ENV,"JULIA_SHELL",get(ENV,"SHELL","/bin/sh")))
-    shell_name = Base.basename(shell[1])
+repl_cmd(line::AbstractString, out) = repl_cmd(Val(Sys.iswindows()), line, out)
 
-    if isempty(cmd.exec)
+function repl_cmd(iswindows::Val{false}, line::AbstractString, out)
+    shell = shell_split(get(ENV, "JULIA_SHELL", get(ENV,"SHELL","/bin/sh")))
+    shell_name = basename(shell[1])
+    cmd = cmd_gen(eval(Main, shell_parse(line)[1]))
+    if isempty(cmd)
         throw(ArgumentError("no cmd to execute"))
-    elseif cmd.exec[1] == "cd"
-        new_oldpwd = pwd()
-        if length(cmd.exec) > 2
-            throw(ArgumentError("cd method only takes one argument"))
-        elseif length(cmd.exec) == 2
-            dir = cmd.exec[2]
-            if dir == "-"
-                if !haskey(ENV, "OLDPWD")
-                    error("cd: OLDPWD not set")
-                end
-                cd(ENV["OLDPWD"])
-            else
-                cd(@static Sys.iswindows() ? dir : readchomp(`$shell -c "echo $(shell_escape(dir))"`))
+    elseif cmd[1] == "cd" && length(cmd) <= 2
+        repl_cd(cmd, shell, out)
+    else
+        run(ignorestatus(isa(STDIN, TTY) ? `$shell -i -c "$(shell_wrap_true(shell_name, cmd))"` : `$shell -c "$(shell_wrap_true(shell_name, cmd))"`))
+    end
+    nothing
+end
+
+function repl_cmd(iswindows::Val{true}, line::AbstractString, out)
```

Comment:
don't use `Val`

Reply:
OK, I'm thinking I will just use a single run-time `if` statement  within a common `repl_cmd` method.

Just for my own edification, why is using `Val` not recommended? Is it a performance issue?
I thought `iswindows()` would be known at compile time?
Are there any cases where using `Val` is OK?